packages/liveblocks-react/src/umbrella-store.ts

```diff
@@ -299,6 +299,72 @@ const noop = Promise.resolve();
 
 const ASYNC_LOADING = Object.freeze({ isLoading: true });
 
+class SingleResource {
+  public readonly observable: Observable<void>;
+  private _eventSource: EventSource<void>;
+  private _fetchPage: () => Promise<void>;
+  private _cachedPromise: UsablePromise<void> | null = null;
```

Comment:
I've duplicated the implementation from `PaginatedResource` for now but we most likely need an abstraction that both single and paginated resources can extends.

Reply:
Yep! Duplication isn't bad here. Premature DRYing up or using subclasses would be much worse. Until the abstraction becomes apparent, we can keep the duplication. If they're still somewhat symmetric, this is fine.